src/damavand/cloud/aws/resources/glue_component.py

```diff
@@ -10,19 +10,100 @@
+           * For the G.1X worker type, each worker maps to 1 DPU (4 vCPU, 16 GB of memory, 64 GB disk), and provides 1 executor per worker. Recommended for memory-intensive jobs.
+           * For the G.2X worker type, each worker maps to 2 DPU (8 vCPU, 32 GB of memory, 128 GB disk), and provides 1 executor per worker. Recommended for memory-intensive jobs.
+           * For the G.4X worker type, each worker maps to 4 DPU (16 vCPUs, 64 GB of memory) with 256GB disk (approximately 235GB free), and provides 1 executor per worker. Recommended for memory-intensive jobs. Only available for Glue version 3.0. Available AWS Regions: US East (Ohio), US East (N. Virginia), US West (Oregon), Asia Pacific (Singapore), Asia Pacific (Sydney), Asia Pacific (Tokyo), Canada (Central), Europe (Frankfurt), Europe (Ireland), and Europe (Stockholm).
+           * For the G.8X worker type, each worker maps to 8 DPU (32 vCPUs, 128 GB of memory) with 512GB disk (approximately 487GB free), and provides 1 executor per worker. Recommended for memory-intensive jobs. Only available for Glue version 3.0. Available AWS Regions: US East (Ohio), US East (N. Virginia), US West (Oregon), Asia Pacific (Singapore), Asia Pacific (Sydney), Asia Pacific (Tokyo), Canada (Central), Europe (Frankfurt), Europe (Ireland), and Europe (Stockholm).
+           * For the G.025X worker type, each worker maps to 0.25 DPU (2 vCPU, 4GB of memory, 64 GB disk), and provides 1 executor per worker. Recommended for low volume streaming jobs. Only available for Glue version 3.0.
+           * For the Z.2X worker type, each worker maps to 2 M-DPU (8vCPU, 64 GB of m emory, 128 GB disk), and provides up to 8 Ray workers based on the autoscaler.
+    :param enable_glue_datacatalog: To use the Glue catalog as the metadata catalog
+    :param enable_continuous_cloudwatch_log:
+    :param enable_continuous_log_filter: When set to true it reduces the amount of logging.
+        For more information see https://repost.aws/knowledge-center/glue-reduce-cloudwatch-logs
+    :param enable_metrics: Enables observability metrics about the worker nodes.
+    :param enable_observability_metrics: Enables extra Spark-related observability metrics such as how long a tasks takes.
+        This parameter could increase cloud costs significantly.
+    :param script_args: The arguments that your own script consumes {"--arg1": "arg1 value"}
+    :param schedule: A time-based cron-like schedule for the job. For syntax rules see https://docs.aws.amazon.com/glue/latest/dg/monitor-data-warehouse-schedule.html
+    :param job_type: Specify if the job is streaming or batch. Possible values: glueetl, gluestreaming
+    """
+    # Parameters for Pulumi Glue Job
     name: str
-    description: str
-    options: dict[str, Any] = field(default_factory=dict)
+    description: str = None
+    script_location: str = None
+    extra_libraries: list[str] = field(default_factory=list)
+    execution_class: str = "STANDARD"
```

Comment:
What are the alternatives? think like you are the end user of this class. The library asking for execution_class, but how you gonna know what are the options? I suggest to create an enum that fallback to string type for both compatibility and sugar syntax. Instead of putting the acceptable values in the docstring you can refer to aws docs for future compatibility.

Reply:
Done